diesel/src/pg/types/floats/mod.rs

```diff
@@ -117,16 +117,18 @@ impl ToSql<sql_types::Numeric, Pg> for PgNumeric {
 impl FromSql<sql_types::Float, Pg> for f32 {
     fn from_sql(value: PgValue<'_>) -> deserialize::Result<Self> {
         let mut bytes = value.as_bytes();
-        debug_assert!(
-            bytes.len() <= 4,
-            "Received more than 4 bytes while decoding \
-             an f32. Was a double accidentally marked as float?"
-        );
-        debug_assert!(
-            bytes.len() >= 4,
-            "Received less than 4 bytes while decoding \
-             an f32."
-        );
+
+        #[cfg(debug_assertions)]
```

Comment:
I would prefer returning a straight error here instead. I doubt that this will actually break anyone. (And if it does it uncovers bugs in their application)

Reply:
Done!